packages/agreement-process/src/services/agreementSubmissionProcessor.ts

```diff
@@ -60,8 +62,9 @@ export const createSubmissionUpdateAgreementSeed = (
   userId: UserId
 ): UpdateAgreementSeed => {
   const stamps = calculateStamps(agreement, newState, createStamp(userId));
+  const isActivation = agreement.state === agreementState.active;
```

Comment:
I prefer to use a variable to explain what meaning this condition

Reply:
It the update from `state` to `newState` happens below, shouldn't you use newState for this check here? `agreement.state` is still the old state in this case.
```suggestion
  const isActivation = newState === agreementState.active;
```